backend/app/repositories/user_repository.py

```diff
@@ -1,36 +1,95 @@
+from typing import Optional, Union
+
 from sqlalchemy import select
-from sqlalchemy.ext.asyncio import async_sessionmaker
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.orm import selectinload
 
-from app.core.db import engine
+from app.core.db import async_session
 from app.models import User
+from app.schemas.user_schema import UserEntity
+
 
-async_session = async_sessionmaker(engine, expire_on_commit=False)
+def _to_user_public(user: User) -> UserEntity:
```

Comment:
when we changed back to `UserEntity` maybe this should be
```suggestion
def _to_user_entity(user: User) -> UserEntity:
```